### ISSUE TITLE
Error: self signed certificate in certificate chain

### ISSUE THREAD

**Issue by johnc1984** [request/request#1764]
I have managed to get the majority of http and https sites I have tested working, however I am having difficulties finding any documentation on the web about this error from 'request':
```
{ [Error: self signed certificate in certificate chain] code: 'SELF_SIGNED_CERT_IN_CHAIN' }
```
 
The URL in question:
**https://esta.cbp.dhs.gov/esta/**

The debug:
```
server-0 (err): REQUEST { url: 'https://esta.cbp.dhs.gov/esta/',
server-0 (err):   headers:
server-0 (err):    { 'User-Agent': 'Mozilla/5.0 (Windows NT 6.3; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/44.0.2403.155 Safari/537.36 OPR/31.0.1889.174',
server-0 (err):      'Accept-Language': 'en-GB,en-US;q=0.8,en;q=0.6',
server-0 (err):      Accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8',
server-0 (err):      'Cache-Control': 'no-cache',
server-0 (err):      Pragma: 'no-cache',
server-0 (err):      Connection: 'keep-alive',
server-0 (err):      DNT: 1,
server-0 (err):      'Accept-Encoding': 'gzip',
server-0 (err):      'Upgrade-Insecure-Requests': 1 },
server-0 (err):   gzip: true,
server-0 (err):   callback: [Function] }
server-0 (err): REQUEST make request https://esta.cbp.dhs.gov/esta/
```

The code
```
//set the options
var options = {
	url: returnJson.inputData.url,
	headers: {
		/* Mimicking the browsers request headers */
		'User-Agent'					: 'Mozilla/5.0 (Windows NT 6.3; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/44.0.2403.155 Safari/537.36 OPR/31.0.1889.174',
		'Accept-Language' 				: 'en-GB,en-US;q=0.8,en;q=0.6',
		'Accept'						: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8',
		'Cache-Control'					: 'no-cache',
		'Pragma'						: 'no-cache',
		'Connection'					: 'keep-alive',
		'DNT'							: 1,
		'Accept-Encoding'				: 'gzip',
		'Upgrade-Insecure-Requests'		: 1
	},
	gzip : true
};

request( options , function (error, response, body) {

	log( error );
});
```

I there another trick I might be missing?

**Comment by simov** [comment]
`strictSSL:false` maybe?

**Comment by cbleeker** [comment]
I had a similar issue.  The request module uses the default node http module on the backend correct?  It seems as if Node ships with its own root CA's and doesn't use the latest.  I used this module: https://github.com/coolaj86/node-ssl-root-cas to inject the latest CA's (even though I updated openssl's, and the Centos server's trusted CA's it had no effect on Node!).  Surprised this isn't a bigger issue...
